tests/v2/validation/snapshot/snapshot.go

```diff
@@ -4,270 +4,351 @@ import (
+		clusterObject, _, err = clusters.GetProvisioningClusterByName(client, clusterName, namespace)
+		require.NoError(t, err)
+
+		steveID := fmt.Sprintf("%s/%s", clusterObject.Namespace, clusterObject.Name)
+		err = clusters.WatchAndWaitForCluster(client, steveID)
+		require.NoError(t, err)
+
+		podErrors := pods.StatusPods(client, clusterID)
+		assert.Empty(t, podErrors)
+
+		initialKubernetesVersion := clusterObject.Spec.KubernetesVersion
+		require.Equal(t, initialKubernetesVersion, clusterObject.Spec.KubernetesVersion)
 	}
-	err = etcdsnapshot.RestoreSnapshot(client, clusterName, snapshotRestore)
-	require.NoError(t, err)
 
-	err = clusters.WaitClusterToBeUpgraded(client, clusterID)
+	snapshotToRestore, err := provisioning.VerifySnapshots(client, localClusterID, clusterName, etcdNodeCount+len(existingSnapshots), isRKE1)
 	require.NoError(t, err)
 
-	podErrors = pods.StatusPods(client, clusterID)
-	assert.Empty(t, podErrors)
+	postBackupDeployment := workloads.NewDeploymentTemplate(WorkloadNamePostBackup, defaultNamespace, podTemplate, isCattleLabeled, nil)
+	postBackupService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAppendName + WorkloadNamePostBackup,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: port,
+					Port: 80,
+				},
+			},
+			Selector: deployment.Spec.Template.Labels,
+		},
+	}
+
+	postDeploymentResp, _, err := workloads.CreateDeploymentWithService(steveclient, WorkloadNamePostBackup, postBackupDeployment, postBackupService)
 	require.NoError(t, err)
-	require.Equal(t, initialKubernetesVersion, clusterObject.Spec.KubernetesVersion)
-
-	// validate restored workload
-	var deploymentList *steveV1.SteveCollection
-	err = kwait.Poll(1*time.Second, 5*time.Minute, func() (done bool, err error) {
-		deploymentList, err = steveclient.SteveType(workloads.DeploymentSteveType).NamespacedSteveClient(defaultNamespace).List(nil)
-		if err != nil {
-			return false, err
-		} else if len(deploymentList.Data) > 0 {
-			return true, nil
-		}
-		return false, nil
-	})
 
+	err = workloads.VerifyDeployment(steveclient, postDeploymentResp)
 	require.NoError(t, err)
-	require.Equal(t, 1, len(deploymentList.Data))
-	require.Equal(t, initialWorkloadName, deploymentList.Data[0].ObjectMeta.Name)
+	require.Equal(t, WorkloadNamePostBackup, postDeploymentResp.ObjectMeta.Name)
 
-	if upgrade {
-		clusterObject, _, err := clusters.GetProvisioningClusterByName(adminClient, clusterName, namespace)
-		require.NoError(t, err)
-		require.Equal(t, initialKubernetesVersion, clusterObject.Spec.KubernetesVersion)
-		if strategy {
-			require.Equal(t, concurrencyDefaultValue, clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency)
-			require.Equal(t, concurrencyDefaultValue, clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency)
-		}
-	}
-}
+	if etcdRestore.SnapshotRestore != none {
```

Comment:
I feel we should validate that the other 2 valid options were entered correctly (this must be `all` or `kubernetesVersion` for the restore to work) at some point near the beginning of this function. Also, if snapshotRestore is not entered/null, will this run?

Reply:
Yeah I considered that and chose to do this. I can make that change tho since it seems to make more sense to you as well the other way.